cpp/modmesh/toggle/RadixTree.hpp

```diff
@@ -265,14 +264,10 @@ class CallProfilerProbe
 {
 public:
     CallProfilerProbe(CallProfiler & profiler, const char * caller_name)
-        : m_profiler(profiler)
-        , m_caller_name(caller_name)
+        : m_caller_name(caller_name)
+        , m_profiler(profiler)
     {
-        auto cancel_callback = [&]()
-        {
-            cancel();
-        };
-        m_profiler.start_caller(m_caller_name, cancel_callback);
```

Comment:
why do you remove the `cancel_ballback`?

Reply:
@tigercosmos 
Because the compiler raised unused variable error.
Also, I am not sure about the true behavior of this callback.
Could you please elaborate more or maybe kindly add some tests in gtests/test_nopython_callprofiler.cpp?

Reply:
@q40603 Originally, `cancel_callback` is for canceling the profiling. For example, if we restart the profiler, we need to cancel the previous profiling, otherwise the profiler will have unexpected behavior. Surprisingly, I didn't fully support the cancel mechanism before. Let me file another PR for this.